vector.cpp

```diff
@@ -0,0 +1,226 @@
+    }
+    cout << endl;
+}
+
+void printVectorArr(const vector<string>& vec) {
+    for (const string& val : vec) {
+        cout << val << " ";
+    }
+    cout << endl;
+}
+
+void getIndex(const vector<int>& vec, int K) {
+    auto it = find(vec.begin(), vec.end(), K);
+    if (it != vec.end()) {
+        int index = it - vec.begin();
+        cout << "Index: " << index << endl;
+    } else {
+        cout << "INDEX NOT FOUND" << endl;
+    }
+}
+
+void getIndex(const vector<string>& vec, const string& K) {
+    auto it = find(vec.begin(), vec.end(), K);
+    if (it != vec.end()) {
+        int index = it - vec.begin();
+        cout << "Index: " << index << endl;
+    } else {
+        cout << "INDEX NOT FOUND" << endl;
+    }
+}
+
+int main() 
+{
+    int opt;
+    int mainNum;
+    vector<int> intVec;
+    vector<string> strVec;
+    bool flag = false;
+    bool mainFlag = false;
+
+    while (!mainFlag) // Is this an issue 
+    {
+        while (!flag) 
+        {
+            cout << "WHICH DATA TYPE WOULD YOU LIKE YOUR VECTOR TO BE?" << endl;
+            cout << "1. INTEGER (TYPE 1 FOR THIS OPTION)" << endl;
+            cout << "2. STRING (TYPE 2 FOR THIS OPTION)" << endl;
+            cin >> opt;
+
+            if (opt == 1) 
+            {
+                int num;
+                int ele = 1;
+                int val;
+                cout << "PLEASE TYPE THE NUMBER OF ELEMENTS YOU WANT IN YOUR VECTOR!" << endl;
+                cin >> num;
+
+                for (int i = 0; i < num; ++i) 
+                {
+                    cout << "PLEASE TYPE YOUR DESIRED INTEGER FOR ELEMENT: " + to_string(ele) << endl;
+                    cin >> val;
+                    intVec.push_back(val);
+                    ele++;
+                }
+
+                mainNum = num; 
+                printVectorArr(intVec);// why does it not print this
+                break;
+        
+            } 
+            else if (opt == 2) 
+            {
+                int num;
+                int ele = 1;
+
+                cout << "PLEASE TYPE THE NUMBER OF ELEMENTS YOU WANT IN YOUR VECTOR!" << endl;
+                cin >> num;
+
+                for (int i = 0; i < num; ++i) 
+                {
+                    string word;
+                    cout << "PLEASE TYPE YOUR DESIRED WORD FOR ELEMENT: " + to_string(ele) << endl;
+                    cin >> word;
+                    strVec.push_back(word);
+                    ele++;
+                }
+
+                mainNum = num;
+                printVectorArr(strVec);// why does it not print this
+                break;
+            } 
+            else 
+            {
+                cout << "****************************************************" << endl;
+                cout << " !!!ERROR PLEASE TRY AGAIN NOT A VALID OPTION!!! " << endl;
+                cout << "****************************************************" << endl;
+                flag = false;// is this wrong?
```

Comment:
It would be an infinite loop right? 

You want to keep taking input as long as flag is false. Flag would be false here also,. Ideally what should be done is 

```
flag = true
while(flag) {
  //do yourstuff
  //set flag = false when you want to come out fro the loop

}
```